packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -152,128 +154,137 @@ const isReactNative = (_textInputRef: any): _textInputRef is TextInputReactNativ
   return getPlatformType() === 'react-native';
 };
 
-export const TextInput = ({
-  label,
-  labelPosition = 'top',
-  placeholder,
-  type = 'text',
-  defaultValue,
-  name,
-  value,
-  maxCharacters,
-  onChange,
-  onFocus,
-  onBlur,
-  isDisabled,
-  necessityIndicator,
-  validationState,
-  errorText,
-  helpText,
-  successText,
-  isRequired,
-  icon,
-  prefix,
-  showClearButton,
-  onClearButtonClick,
-  isLoading,
-  suffix,
-  autoFocus,
-  keyboardReturnKeyType,
-  autoCompleteSuggestionType,
-}: TextInputProps): ReactElement => {
-  const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
-  const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+export const TextInput = React.forwardRef<NativeHTMLInputRefProps, TextInputProps>(
```

Comment:
nit: just for the sake of readability can we keep this declaration of component to how it was before? and just call forwardRef at the end? like below

```
const _TextInput = ({props},ref) => ()

export const TextInput = React.forwardRef(_TextInput,ref)
```